pkg/install/util/crd.go

```diff
@@ -70,6 +71,34 @@ func DeployCRDs(ctx context.Context, kubeClient ctrlruntimeclient.Client, log lo
 	return nil
 }
 
+// DeleteOldApplicationInstallationCrd removes old applicationInstallation crd from cluster.
+// See pkg/install/stack/kubermatic-master/stack.go::InstallKubermaticCRDs() for more information.
+// TODO REMOVE AFTER release v2.21.
+func DeleteOldApplicationInstallationCrd(ctx context.Context, kubeClient ctrlruntimeclient.Client) error {
+	crd := &apiextensionsv1.CustomResourceDefinition{}
+	err := kubeClient.Get(ctx, types.NamespacedName{Name: appskubermaticv1.ApplicationInstallationsFQDNName}, crd)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to get CRD %s: %w", appskubermaticv1.ApplicationInstallationsFQDNName, err)
+	}
+
+	// No action is required
+	if err != nil && apierrors.IsNotFound(err) {
+		return nil
+	}
+
+	// CRD exists, now we need to check if it's cluster scoped
+	if crd.Spec.Scope == apiextensionsv1.ClusterScoped {
```

Comment:
This won't work, because when the CRD is on 2.21 (i.e. namespaced), you're code still waits for it to be gone. Let's just exit early with

```go
if crd.Spec.Scope == apiextensionsv1.NamespaceScope {
    return nil
}
```